Source/WebKit/WebProcess/Cache/WebCacheStorageConnection.cpp

```diff
@@ -77,7 +77,16 @@ void WebCacheStorageConnection::retrieveCaches(const WebCore::ClientOrigin& orig
 
 void WebCacheStorageConnection::retrieveRecords(WebCore::DOMCacheIdentifier cacheIdentifier, WebCore::RetrieveRecordsOptions&& options, WebCore::DOMCacheEngine::RecordsCallback&& callback)
 {
-    connection().sendWithAsyncReply(Messages::NetworkStorageManager::CacheStorageRetrieveRecords(cacheIdentifier, options), WTFMove(callback));
+    auto newCallback = [callback = WTFMove(callback)](auto&& result) mutable {
```

Comment:
I am not sure we want this.
Since we have cross thread records, we could optimise worker connection code by doing a simple move or at least an optimised isolatedCopy()&&?